pkg/config/remote/service/service.go

```diff
@@ -349,8 +371,12 @@ func NewService(cfg model.Reader, rcType, baseRawURL, hostname string, tagsGette
 
 	clock := clock.New()
 
-	return &Service{
-		rcType:                         rcType,
+	return &CoreAgentService{
+		Service: Service{
+			rcType: rcType,
+			uptane: uptaneClient,
+			db:     db,
+		},
```

Comment:
Something still feels a little awkward here even with the revert of that one refactor. For both the CoreAgentService and the HTTPClient the `uptaneClient` in the Service is the same as the client used in the containing struct. I think I see the abstraction layer we are trying to draw, but the fact that the identical reference to the client is in Service -and- CoreAgentService feels like maybe the boundaries aren't quite right. Maybe the methods of Service could accept an uptaneClient as a parameter?

Reply:
Agreed - I think there are still more improvements that could be made but for now I have added uptaneClient as a parameter to the service methods as you suggested, which is much nicer